src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1473,6 +1473,20 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        private bool DetectRecursiveSymLink(DirectoryInfo path)
+        {
+            var readSymLink = Platform.NonWindowsInternalGetTarget(path.FullName);
+
+            if (path.FullName.Contains(readSymLink))
```

Comment:
Maybe `path.FullName.StartsWith` is a better check. Considering the following case:
```
path = /tmp/home/pi/foo
readSymLink = /home/pi
```